internal/functions/Compare-DbaStringCollation.ps1

```diff
@@ -0,0 +1,130 @@
+        https://docs.microsoft.com/en-us/dotnet/api/system.string.compare?view=net-5.0
+    .EXAMPLE
+        PS C:\> Compare-DbaStringCollation -Reference dbatools -Difference DbAToOlS -Comparison Eq -Collation SQL_Latin1_General_CP1_CI_AS
+        Returns
+        Collation  : SQL_Latin1_General_CP1_CI_AS
+        Reference  : dbatools
+        Difference : DbAToOlS
+        Comparison : eq
+        Equivalent : True
+    .EXAMPLE
+        PS C:\> Compare-DbaStringCollation -Reference Dbátools -Difference Dbatools -Comparison Eq -Collation SQL_Latin1_General_CP1_CS_AI
+        Returns
+        Collation  : SQL_Latin1_General_CP1_CS_AI
+        Reference  : Dbátools
+        Difference : Dbatools
+        Comparison : Eq
+        Equivalent : True
+    .Example
+         PS C:\> Compare-DbaStringCollation -Reference dbátools -Difference DbAtOoLs -Comparison Ne -Collation SQL_Latin1_General_CP1_CI_AI
+         Returns
+         Collation  : SQL_Latin1_General_CP1_CI_AI
+         Reference  : dbátools
+         Difference : DbAtOoLs
+         Comparison : Ne
+         Equivalent : False
+    .EXAMPLE
+        PS C:\> $collations =  Get-DbaAvailableCollation -SqlInstance localhost
+        PS C:\> Compare-DbaStringCollation -Reference ﾐﾑﾒ -Difference ミムメ -Comparison Eq -Collation $collations.Name
+        Returns all the collations available on SqlInstance localhost and how ﾐﾑﾒ and ミムメ would evaluate for each of them
+    .EXAMPLE
+        PS C:\> Compare-DbaStringCollation -Reference dbá,Tools -Difference Dba,tools -Comparison In -Collation SQL_Latin1_General_CP1_CI_AI
+        Returns
+        Collation  : SQL_Latin1_General_CP1_CI_AI
+        Reference  : dbá
+        Difference : {Dba, tools}
+        Comparison : In
+        Equivalent : True
+
+        Collation  : SQL_Latin1_General_CP1_CI_AI
+        Reference  : Tools
+        Difference : {Dba, tools}
+        Comparison : In
+        Equivalent : True
+    .EXAMPLE
+        PS C:\> Compare-DbaStringCollation -Reference dbá,Tools -Difference Dba,tools -Comparison Notin -Collation SQL_Latin1_General_CP1_CI_AS
+
+        Collation  : SQL_Latin1_General_CP1_CI_AS
+        Reference  : dbá
+        Difference : {Dba, Tools}
+        Comparison : Notin
+        Result     : True
+
+        Collation  : SQL_Latin1_General_CP1_CI_AS
+        Reference  : tools
+        Difference : {Dba, Tools}
+        Comparison : Notin
+        Result     : False
+    #>
+    param(
+        [Parameter(Mandatory)]
+        [string[]]$Reference,
+        [Parameter(Mandatory)]
+        $Difference,
+        [Parameter(Mandatory)]
+        [string[]]$Collation,
+        [Parameter(Mandatory)]
+        [ValidateSet('In', 'Notin', 'Eq', 'Ne')]
+        [string]$Comparison)
+    $smo = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Server
+
+    $results = @()
+    foreach ($col in $Collation) {
+        foreach ($ref in $Reference) {
+            switch ($Comparison) {
```

Comment:
Instead of doing a switch on each object in `$Reference` I'd recommend having the foreach work from within the switch blocks, if they are needed.

Something like:

```powershell
switch ($Comparison) {
    'Eq' {
        foreach ($ref in $Reference) {
            ...
    }
    ...
}
```

Reply:
trying

but haven't used labeled breaks before in powershell, and it's starting to get confusing

```suggestion
            switch ($Comparison) {
            ...
             foreach ( $ref in $Reference) {
                $res = $null
                :fordifference foreach ($dif in $Difference) {
                    $testCondition = $smo.getStringComparer($Collation).Compare($ref, $dif)
                    if ($testCondition -eq 0 ) {
                        $res
                        $results += [pscustomobject]@{Collation = $Collation
                            Reference                           = $ref
                            Difference                          = $Difference
                            Comparison                          = $Comparison
                            Result                              = $false
                        }
                        break :fordifference
                    }
                }
            }
            if ($null -eq $res ) {
                $results += [pscustomobject]@{Collation = $Collation
                    Reference                           = $ref
                    Difference                          = $Difference
                    Comparison                          = $Comparison
                    Result                              = $true
                }
            }
        }
    }
```

Reply:
Why would you need to break?

Reply:
With `foreach` blocks we do not gather the output in a collection and output as it is processed.

Reply:
Original Implementation was just, return boolean, so it relied on that exit properly when a Match was found. 

last pull request worked the same way, but used break to exit the switch when a match was found. 

Trying to get the suggestion to move the foreach into each of the switch statements to work, but it no longer does.

Reply:
```suggestion
```
Remove Unused Switch